src/main/kotlin/org/rust/lang/core/types/visitors/impl/RustTypificationEngine.kt

```diff
@@ -217,7 +217,7 @@ private class RustTypeTypificationVisitor(val pivot: RsType) : RustComputingVisi
             RustUnitType
     }
 
-    override fun visitBaseType(o: RsBaseType) = set {
+    override fun visitBaseType(o: RsBaseType): Unit = o.baseType?.let { visitBaseType(it) } ?: set {
```

Comment:
I made it inside at first, but than decided to extract for cleaner code.
I can return it back.